subgraphs/euler-finance/src/mappings/helpers.ts

```diff
@@ -162,316 +139,607 @@ export function createRepay(event: Repay): BigDecimal {
-    const reserveYield = reserveAPY.times(secondsSinceLastUpdate.toBigDecimal()).div(SECONDS_PER_YEAR);
-    const protocolSideRevenueSinceLastUpdate = reserveYield.times(market.totalDepositBalanceUSD);
+  log.debug(
+    "[updateInterestRates]interest rates for market {} updated to [{}, {}] at block {}; inputs:interestRate={},reserveFee={},totalBorrows={},totalBalances={},tx={}",
+    [
+      market.id,
+      borrowerRate.rate.toString(),
+      lenderRate.rate.toString(),
+      event.block.number.toString(),
+      interestRate.toString(),
+      reserveFee.toString(),
+      totalBorrows.toString(),
+      totalBalances.toString(),
+      event.transaction.hash.toHexString(),
+    ],
+  );
 
-    const totalRevenueSinceLastUpdate = supplySideRevenueSinceLastUpdate.plus(protocolSideRevenueSinceLastUpdate);
+  const marketDailySnapshot = getOrCreateMarketDailySnapshot(event.block, market.id);
+  const days = (event.block.timestamp.toI32() / SECONDS_PER_DAY).toString();
+  marketDailySnapshot.rates = getSnapshotRates(market.rates, days);
+  marketDailySnapshot.blockNumber = event.block.number;
+  marketDailySnapshot.timestamp = event.block.timestamp;
+  marketDailySnapshot.save();
 
-    // update protocol revenue
-    protocol.cumulativeSupplySideRevenueUSD = protocol.cumulativeSupplySideRevenueUSD.plus(
-      supplySideRevenueSinceLastUpdate,
-    );
-    protocol.cumulativeProtocolSideRevenueUSD = protocol.cumulativeProtocolSideRevenueUSD.plus(
-      protocolSideRevenueSinceLastUpdate,
-    );
-    protocol.cumulativeTotalRevenueUSD = protocol.cumulativeTotalRevenueUSD.plus(totalRevenueSinceLastUpdate);
+  const marketHourlySnapshot = getOrCreateMarketDailySnapshot(event.block, market.id);
+  const hours = (event.block.timestamp.toI32() / SECONDS_PER_DAY).toString();
+  marketHourlySnapshot.rates = getSnapshotRates(market.rates, hours);
+  marketHourlySnapshot.blockNumber = event.block.number;
+  marketHourlySnapshot.timestamp = event.block.timestamp;
+  marketHourlySnapshot.save();
+}
 
-    // update market's revenue
-    market.cumulativeSupplySideRevenueUSD = market.cumulativeSupplySideRevenueUSD.plus(
-      supplySideRevenueSinceLastUpdate,
-    );
-    market.cumulativeProtocolSideRevenueUSD = market.cumulativeProtocolSideRevenueUSD.plus(
-      protocolSideRevenueSinceLastUpdate,
-    );
-    market.cumulativeTotalRevenueUSD = market.cumulativeTotalRevenueUSD.plus(totalRevenueSinceLastUpdate);
-    market.save();
+export function updateRevenue(
+  reserveBalance: BigInt,
+  totalBalances: BigInt,
+  totalBorrows: BigInt,
+  protocol: LendingProtocol,
+  market: Market,
+  assetStatus: _AssetStatus,
+  event: ethereum.Event,
+): void {
+  const marketId = market.id;
+  const underlying = Address.fromString(market.inputToken);
+  const block = event.block;
+  const timestamp = event.block.timestamp;
+  const token = getOrCreateToken(underlying);
+
+  if (assetStatus.timestamp && timestamp.lt(assetStatus.timestamp)) {
```

Comment:
is there a way we can handle this without halting indexing?

Reply:
It is unlikely to happen, I am deleting them.